edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/NestedDeviceScopeApiClient.cs

```diff
@@ -116,12 +118,12 @@ static Task<T> ExecuteWithRetry<T>(Func<Task<T>> func, Action<RetryingEventArgs>
             return transientRetryPolicy.ExecuteAsync(func);
         }
 
-        async Task<ScopeResult> GetIdentityOnBehalfOfWithRetry(Uri uri, string deviceId, Option<string> moduleId)
+        async Task<ScopeResult> GetIdentityOnBehalfOfWithRetry(Uri uri, string deviceId, Option<string> moduleId, string onBehalfOfDevice)
```

Comment:
Nit:This method is only called from GetIdentityOnBehalfOfAsync.  Suggest to combine.

Reply:
yea it's a little weird, but it seems to be a pattern from the legacy device scope client implementation, so I opted to just keep it as is for consistency.

---
In reply to: [479387562](https://github.com/Azure/iotedge/pull/3467#discussion_r479387562) [](ancestors = 479387562)